src/main/java/com/lineate/testyourlexicon/controllers/GameController.java

```diff
@@ -40,17 +45,22 @@ public SupportedLanguagesDto supportedLanguages() {
 
   @PutMapping("/configuration")
   public GameConfigurationDto configure(
-      @RequestBody @Valid GameConfigurationDto gameConfigurationDto) {
-    return gameService.configure(gameConfigurationDto, getAuthenticatedUser());
+      @RequestBody @Valid GameConfigurationDto gameConfigurationDto,
+      HttpServletRequest request) {
+    return gameService.configure(gameConfigurationDto, getUserHash(request));
   }
 
   @GetMapping("/configuration")
-  public GameConfigurationDto configuration() {
-    return gameService.userConfiguration(getAuthenticatedUser());
+  public GameConfigurationDto configuration(HttpServletRequest request) {
+    return gameService.userConfiguration(getUserHash(request));
   }
 
-  private User getAuthenticatedUser() {
-    return authenticationService.getAuthenticatedUser()
-      .orElseThrow(UserNotAuthenticatedException::new);
+  private Long getUserHash(HttpServletRequest request) {
+    if (authenticationService.isAuthenticated()) {
+      Long id = userService
+          .findUserByEmail(authenticationService.getAuthenticatedUserEmail()).getId();
+      return Hash.hashToLong(id);
+    }
+    return Hash.hashToLong(request.getSession().getId());
```

Comment:
Let's keep logic out of the controller. We can put this method into AuthenticationService and instead of HttpServletRequest pass sessionId as a parameter.